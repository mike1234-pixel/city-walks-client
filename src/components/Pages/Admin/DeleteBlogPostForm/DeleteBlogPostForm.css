/** @format */
.delete-blog-post-form {
  padding-top: 20px;
  padding-bottom: 20px;
}

/*# sourceMappingURL=DeleteBlogPostForm.css.map */
