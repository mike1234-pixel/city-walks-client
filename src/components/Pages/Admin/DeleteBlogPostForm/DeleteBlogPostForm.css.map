{"version":3,"sourceRoot":"","sources":["DeleteBlogPostForm.scss"],"names":[],"mappings":"AAAA;AAEA;EACE;EACA","file":"DeleteBlogPostForm.css"}