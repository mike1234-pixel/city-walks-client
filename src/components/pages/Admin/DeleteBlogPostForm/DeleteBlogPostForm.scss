/** @format */

.delete-blog-post-form {
  padding: 20px;
}
