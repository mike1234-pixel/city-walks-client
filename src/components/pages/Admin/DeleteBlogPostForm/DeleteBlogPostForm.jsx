import { useState } from "react"
import { MDBInput, MDBBtn, MDBIcon } from "mdbreact"
import axios from "axios"
import toTitleCase from '../../../../functions/toTitleCase'
import './DeleteBlogPostForm.css'

const DeleteBlogPostForm = () => {

    const [postTitle, setPostTitle] = useState("")

    const handleChange = (event) => {
        switch(event.target.name) {
            case "post-title":
              setPostTitle(event.target.value)
              break;
          } 
    }

    const handleSubmit = (event) => {
        event.preventDefault()

        let payload = {
            postTitle: toTitleCase(postTitle)
          };

        axios
        .delete("http://localhost:5000/delete-blog-post",{ data: payload } )
        .then((res, err) => {
          if (err) {
            console.log(err);
          } else if (res.data === "blog post deleted") {
            console.log("blog post deleted")
          } else {
            console.log("response not deleted")
          }
        });

        alert("Blog Post Deleted")
        setPostTitle("")
        window.scrollTo(0, 0);
    }

    return (
    <div>
        <h2>Delete Blog Post</h2>
        <form onSubmit={handleSubmit} className="delete-blog-post-form">
            <MDBInput type="text" name="post-title" id="post-title" value={postTitle} label="blog post title" onChange={handleChange} required/>
            <MDBBtn outline color="elegant" type="submit">
                Delete Post <MDBIcon icon="trash"/>
          </MDBBtn>
        </form>
    </div>
    )
}

export default DeleteBlogPostForm