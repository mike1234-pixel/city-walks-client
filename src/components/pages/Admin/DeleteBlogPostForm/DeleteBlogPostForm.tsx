import React, { useState } from "react"
import { MDBInput, MDBBtn, MDBIcon } from "mdbreact"
import axios, { AxiosResponse } from "axios"
import toTitleCase from '../../../../functions/toTitleCase'
import SelectedPost from "../../../../types/PostRequests/SelectedPost"
import './DeleteBlogPostForm.scss'

const DeleteBlogPostForm: React.FC = () => {

  const [postTitle, setPostTitle] = useState<string>("")

  const handleChange: (event: React.ChangeEvent<any>) => void = (event) => {
    switch (event.target.name) {
      case "post-title":
        setPostTitle(event.target.value)
        break;
    }
  }

  const handleSubmit: (event: React.FormEvent) => void = (event) => {
    event.preventDefault()

    let payload: SelectedPost = {
      postTitle: toTitleCase(postTitle)
    };

    axios
      .delete("https://city-walks.herokuapp.com/delete-blog-post", { data: payload })
      .then((res: AxiosResponse) => {
        if (res.data === "blog post deleted") {
          console.log("blog post deleted")
        } else {
          console.log("response not deleted")
        }
      });

    alert("Blog Post Deleted")
    setPostTitle("")
    window.scrollTo(0, 0);
  }

  return (
    <div>
      <h2>Delete Blog Post</h2>
      <form onSubmit={handleSubmit} className="delete-blog-post-form">
        <MDBInput type="text" name="post-title" id="post-title" value={postTitle} label="blog post title" onChange={handleChange} required />
        <MDBBtn outline color="elegant" type="submit">
          Delete Post <MDBIcon icon="trash" />
        </MDBBtn>
      </form>
    </div>
  )
}

export default DeleteBlogPostForm