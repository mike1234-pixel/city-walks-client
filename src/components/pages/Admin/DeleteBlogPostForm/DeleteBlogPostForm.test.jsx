import renderer from 'react-test-renderer'
import DeleteBlogPostForm from "./DeleteBlogPostForm"

// snapshot test
it('DeleteBlogPostForm Page renders correctly', () => {  

  const tree = renderer
    .create(
        <DeleteBlogPostForm />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});