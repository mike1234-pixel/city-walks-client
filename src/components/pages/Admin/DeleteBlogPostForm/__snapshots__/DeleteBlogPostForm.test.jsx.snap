// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteBlogPostForm Page renders correctly 1`] = `
<div>
  <h2>
    Delete Blog Post
  </h2>
  <form
    className="delete-blog-post-form"
    onSubmit={[Function]}
  >
    <div
      className="md-form"
    >
      
      <input
        aria-disabled={false}
        className="form-control"
        data-test="input"
        disabled={false}
        id="post-title"
        name="post-title"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onInput={[Function]}
        required={true}
        type="text"
        value=""
      />
      <label
        aria-labelledby=""
        className=""
        data-error=""
        data-success=""
        htmlFor="post-title"
        id=""
        onClick={[Function]}
      >
        blog post title
      </label>
    </div>
    <button
      className="btn-elegant btn-outline-elegant btn Ripple-parent"
      data-test="button"
      onMouseUp={[Function]}
      onTouchStart={[Function]}
      type="submit"
    >
      Delete Post 
      <i
        className="fa fa-trash"
        data-test="fa"
      />
      <div
        className="Ripple Ripple-outline "
        data-test="waves"
        style={
          Object {
            "height": "0px",
            "left": "0px",
            "top": "0px",
            "width": "0px",
          }
        }
      />
    </button>
  </form>
</div>
`;
